apache/incubator-zeppelin
# [ZEPPELIN-619] Shared Resource pool across interpreter processes

### What is this PR for?
This is sub task of https://issues.apache.org/jira/browse/ZEPPELIN-533.
It provides shared resource pool to exchange data across interpreter processes.

### What type of PR is it?
Feature


### Is there a relevant Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-619

### How should this be tested?
create two different spark interpreter settings. 
create two different notebooks each bind different spark interpreter setting.

put an object from one notebook.
read the object from the other notebook. (from the other interpreter process)

See screenshot

### Screenshots (if appropriate)
![resource_pool](https://cloud.githubusercontent.com/assets/1540981/12409095/db4e142e-be1b-11e5-91e2-4b91c5b39dc3.gif)



### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? javadoc included.
